src/Compilers/CSharp/Portable/FlowAnalysis/AbstractFlowPass.cs

```diff
@@ -2404,7 +2408,7 @@ private void VisitBinaryOperatorChildren(BoundBinaryOperator node)
                 stack.Push(binary);
                 binary = binary.Left as BoundBinaryOperator;
             }
-            while (binary != null && !binary.OperatorKind.IsLogical());
+            while (binary != null && !binary.OperatorKind.IsLogical() && binary.InterpolatedStringHandlerData is null);
```

Comment:
Do we have a test that is affected by this change?

Reply:
What was the resolution?

Reply:
This is the same test that addressed the nested binary operator question: `InterpolatedStringsAddedUnderObjectAddition`. That will hit all the codepaths that need to break out of a nested binary operation, both in flow analysis and rewriting.

Reply:
I've run the interpolated and nullable tests without this change and none was affected. I'm fine to move ahead (the change seems correct) but please try to find a way to test it.

Reply:
Added a definite assignment test that will fail on this codepath.